### PR TITLE
Restore the configuration after activity resumed

### PR DESCRIPTION
**Pull request by bluk**
- Fix for https://github.com/braintree/braintree_android/issues/181

### CODE REVIEW

File: Drop-In/src/main/java/com/braintreepayments/api/dropin/BaseActivity.java [hunk 1]
```diff
@@ -25,9 +29,23 @@
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
+        if (savedInstanceState != null) {
+            try {
+                mConfiguration = Configuration.fromJson(savedInstanceState.getString(EXTRA_CONFIGURATION_DATA));
+            } catch (JSONException e) {}
```
Comment (sdcoffey) [inline Drop-In/src/main/java/com/braintreepayments/api/dropin/BaseActivity.java]: rename `e` to `ignored` so Android studio won't complain about an empty catch block.

File: Drop-In/src/test/java/com/braintreepayments/api/dropin/BaseActivityUnitTest.java [hunk 2]
```diff
@@ -47,6 +50,65 @@ public void onCreate_setsDropInRequest() {
...
+        Bundle savedState = new Bundle();
+        mActivityController = Robolectric.buildActivity(BaseActivity.class).create(savedState);
+        mActivity = (BaseActivity) mActivityController.get();
+
+        Configuration restoredConfiguration = (Configuration)ReflectionHelper.getField(mActivity, "mConfiguration");
+        assertNull(restoredConfiguration);
+    }
+
+    @Test
+    public void onSaveInstanceState_savesConfigurationAsJSON() throws NoSuchFieldException, IllegalAccessException, JSONException {
```
Comment (quinnjn) [inline Drop-In/src/test/java/com/braintreepayments/api/dropin/BaseActivityUnitTest.java]: `AsJSON` -> `AsJson`

File: Drop-In/src/main/java/com/braintreepayments/api/dropin/BaseActivity.java [hunk 3]
```diff
@@ -14,8 +14,12 @@
 import com.braintreepayments.api.models.Configuration;
 import com.braintreepayments.api.models.PaymentMethodNonce;
 
+import org.json.JSONException;
+
 public class BaseActivity extends AppCompatActivity {
 
+    static final String EXTRA_CONFIGURATION_DATA = "com.braintreepayments.api.EXTRA_CONFIGURATION_DATA";
```
Comment (lkorth) [inline Drop-In/src/main/java/com/braintreepayments/api/dropin/BaseActivity.java]: Any reason this shouldn't be private?

File: Drop-In/src/test/java/com/braintreepayments/api/dropin/BaseActivityUnitTest.java [hunk 4]
```diff
@@ -47,6 +50,65 @@ public void onCreate_setsDropInRequest() {
...
+        Configuration configuration = new TestConfigurationBuilder().buildConfiguration();
+
+        Bundle savedState = new Bundle();
+        savedState.putString(BaseActivity.EXTRA_CONFIGURATION_DATA, configuration.toJson());
+        mActivityController = Robolectric.buildActivity(BaseActivity.class);
+
+        mActivityController.create(savedState);
+        mActivity = (BaseActivity) mActivityController.get();
+
+        Configuration restoredConfiguration = (Configuration)ReflectionHelper.getField(mActivity, "mConfiguration");
```
Comment (lkorth) [inline Drop-In/src/test/java/com/braintreepayments/api/dropin/BaseActivityUnitTest.java]: Space after the cast.

**Pull request merged**